src/tiered/tiered_handle.c

```diff
@@ -301,7 +301,9 @@ __wt_tiered_set_metadata(WT_SESSION_IMPL *session, WT_TIERED *tiered, WT_ITEM *b
 {
     uint32_t i;
 
-    WT_RET(__wt_buf_catfmt(session, buf, ",last=%" PRIu32 ",oldest=%" PRIu32 ",tiers=(",
+    __wt_seconds(session, &tiered->last_flush);
```

Comment:
`wt_tiered_set_metadata()` is called during checkpoint -- from `wt_meta_ckptlist_set()`. Will this will have the effect erroneously reseting the `last_flush` time?

Reply:
It looks like we can't rely on `wt_seconds()` to provide monotonically increasing timestamps across threads.  See this comment in `wt_epoch()`

```
    /*
     * Read into a local variable, then check for monotonically increasing time, ensuring single
     * threads never see time move backward. We don't prevent multiple threads from seeing time move
     * backwards (even when reading time serially, the saved last-read time is per thread, not per
     * timer, so multiple threads can race the time). Nor do we prevent multiple threads
     * simultaneously reading the time from seeing random time or time moving backwards (assigning
     * the time structure to the returned memory location implies multicycle writes to memory).
     */
```

So it seems we could have a flush that happens after a checkpoint in wall-clock-time, but sees an earlier value from `wt_seconds` than the checkpoint.

Reply:
Good catch. I fixed the first bug and changed things around for the second issue. There is code in checkpoint that checks for the clock and I copied that sort of.